CoreWiki/Models/Article.cs

```diff
@@ -9,27 +9,32 @@
 
 namespace CoreWiki.Models
 {
-	public class Article
-	{
-		[Required, Key]
-		public string Topic { get; set; }
+    public class Article
+    {
+        [Key]
```

Comment:
Why do we need an ID field?  This feels unnecessary

Reply:
The reasons I put an Id field in there was:

1. Using the Topic as a key was already broken. There was no check to see it the topic already existed in the database. If you submitted a second topic with the same name as something already in the database you would get the error below:
`SqliteException: SQLite Error 19: 'UNIQUE constraint failed: Articles.Topic'.`

2. Sometime Topics might not be unique. For example: You might use the same "What's new" topic in multiple pages / categories.

3. Database 101 - Primary key's should never change (Id's never change, but Topics do!).

The advantage of having a key that never changes is that you do not have to keep updating other related tables as the key never changes. Id's are better forigen keys for related tables..

**EXAMPLE 1:**
If you use unique Id's (int, guid's etc.) & you have a foreign key on another table, you never have to update the other tables foreign key as it never changes.

Article Table:
--------------
Id = 1
Topic = "Hello World"

ArticleComments Table:
--------------------------
ArticleCommentId = 1
Comment = "Great Article"
ArticleId = 1

**EXAMPLE 2:**
However, if you use the Topic name as your key & you have the Topic name as foreign key on another table, everytime you change the Topic name you have to uodate evryother table that relates to it. What a pain!

Article Table:
--------------
Topic = "Hello World"

ArticleComments Table:
--------------------------
Comment = "Great Article"
ArticleId = "Hello World"

**Just one other quick note regarding id's:**
You could also use the id as part of the url like stackoverflow does. This way the url slug can be anything and does not have to be unique. As the url has the id in it you can always redirect the user to the correct url or use a canonical url like stackoverflow does :)

Example - this url:
https://stackoverflow.com/questions/25259/how-does-stack-overflow-generate-its-seo-friendly-urls

can also be this:
https://stackoverflow.com/questions/25259/hello-world